ENIGMAsystem/SHELL/Universal_System/fileio.cpp

```diff
@@ -220,7 +220,7 @@ int file_bin_open(string fname,int mode) // Opens the file with the indicated na
 bool file_bin_rewrite(int fileid) // Rewrites the file with the given file id, that is, clears it and starts writing at the start.
 {
   enigma::openFile &mf = enigma::files[fileid];
-  mf.f = freopen (mf.sdata.c_str(), "wb", mf.f);
+  mf.f = freopen (mf.sdata.c_str(), "wb+", mf.f);
```

Comment:
Here is a separate test I made and ran in GM8.1. This test proves that in GM8.1 the rewrite function does change the mode regardless of what mode the binary file was originally opened in. In both messages below, I see the number 23 in each. If we comment out the rewrite when the file is opened either mode we get one of two errors.

```cpp
var file_path;
file_path = "C:/Users/Owner/Desktop/test.bin";

var file;
file = file_bin_open("C:/Users/Owner/Desktop/test.bin", 0);
file_bin_rewrite(file); // changes mode from read to readwrite
file_bin_write_byte(file, 23);
file_bin_seek(file,0);
show_message(string(file_bin_read_byte(file)));
file_bin_close(file);

file = file_bin_open("C:/Users/Owner/Desktop/test.bin", 1);
file_bin_rewrite(file); // changes mode from write to readwrite
file_bin_write_byte(file, 23);
file_bin_seek(file,0);
show_message(string(file_bin_read_byte(file)));
file_bin_close(file);
```

Reply:
Alright: 8cdd7e5a99500780973128421757e6eda64bea51

I decided what the heck, let's add a test to cover all of the above too. I like being thorough.

Reply:
I should mention I ran the above in GMSv1.4 and apparently they changed the behavior of rewrite to not do this. Excellent case for applying the compatibility setting, though I am not too concerned about it really.